### PR TITLE
Make ScanOptions optional

### LINKED ISSUE #886: Make ScanOptions optional
# Opening a new issue
___

## Specify type:
- Enhancement

## Description:
Currently the Scan serializer creates default options.  Modify code so it tolerates no options.

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #886

### CODE REVIEW

File: quipucords/api/scan/serializer.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ class DisableOptionalProductsOptionsSerializer(NotEmptySerializer):
     class Meta:
         """Metadata for serializer."""
 
-        model = ExtendedProductSearchOptions
+        model = DisableOptionalProductsOptions
```
Comment (chambridge) [inline quipucords/api/scan/serializer.py]: wow that as a miss from the original PR

File: quipucords/api/scan/model.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ class ScanOptions(models.Model):
     JBOSS_BRMS_EXT = 'jboss_brms_ext'
 
     max_concurrency = models.PositiveIntegerField(default=50)
-    disable_optional_products = \
+    disabled_optional_products = \
         models.ForeignKey(DisableOptionalProductsOptions,
```
Comment (chambridge) [inline quipucords/api/scan/model.py]: Not gonna update this class name, DisableOptionalProductsOptions?

File: quipucords/api/scan/serializer.py (hunk 1 above)
Comment (chambridge) [reply quipucords/api/scan/serializer.py]: DisableOptionalProductsOptions --> DisabledOptionalProductsOptions?

File: quipucords/api/scan/serializer.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ class ScanOptionsSerializer(NotEmptySerializer):
     """Serializer for the ScanOptions model."""
 
     max_concurrency = IntegerField(required=False, min_value=1, default=50)
-    disable_optional_products = DisableOptionalProductsOptionsSerializer(
+    disabled_optional_products = DisableOptionalProductsOptionsSerializer(
```
Comment (chambridge) [inline quipucords/api/scan/serializer.py]: DisableOptionalProductsOptionsSerializer --> DisabledOptionalProductsOptionsSerializer?

File: quipucords/api/scan/model.py (hunk 2 above)
Comment (kholdaway) [reply quipucords/api/scan/model.py]: Done.  All reference updated.

File: quipucords/api/scan/serializer.py (hunk 3 above)
Comment (kholdaway) [reply quipucords/api/scan/serializer.py]: Done.  All reference updated.

File: quipucords/api/scan/serializer.py (hunk 1 above)
Comment (kholdaway) [reply quipucords/api/scan/serializer.py]: Done.  All reference updated.

**Pull request merged**